packages/react-native-sdk/src/components/Call/CallContent/CallContent.tsx

```diff
@@ -201,67 +197,82 @@ export const CallContent = ({
-              <FloatingParticipantView
-                participant={
-                  isRemoteParticipantInFloatingView
-                    ? remoteParticipants[0]
-                    : localParticipant
-                }
-                onPressHandler={handleFloatingViewParticipantSwitch}
-                supportedReactions={supportedReactions}
-                {...participantViewProps}
-              />
+            <View
+              style={[styles.view, callContent.topContainer]}
+              // "box-none" disallows the container view to be not take up touches
+              // and allows only the top and floating view (its child views) to take up the touches
+              pointerEvents="box-none"
+            >
+              {!isInPiPMode && CallTopView && (
+                <CallTopView onBackPressed={onBackPressed} />
+              )}
+              {showFloatingView && FloatingParticipantView && (
+                <FloatingParticipantView
+                  participant={
+                    isRemoteParticipantInFloatingView
+                      ? remoteParticipants[0]
+                      : localParticipant
+                  }
+                  onPressHandler={handleFloatingViewParticipantSwitch}
+                  supportedReactions={supportedReactions}
+                  {...participantViewProps}
+                />
+              )}
+            </View>
+            {showSpotlightLayout ? (
+              <CallParticipantsSpotlight {...callParticipantsSpotlightProps} />
+            ) : (
+              <CallParticipantsGrid {...callParticipantsGridProps} />
             )}
           </View>
-          {showSpotlightLayout ? (
-            <CallParticipantsSpotlight {...callParticipantsSpotlightProps} />
-          ) : (
-            <CallParticipantsGrid {...callParticipantsGridProps} />
-          )}
-        </View>
 
-        {!isInPiPMode && CallControls && (
-          <CallControls
-            onHangupCallHandler={onHangupCallHandler}
-            landscape={landscape}
-          />
-        )}
+          {!isInPiPMode && CallControls && (
+            <CallControls
+              onHangupCallHandler={onHangupCallHandler}
+              landscape={landscape}
+            />
+          )}
+        </SafeAreaView>
       </View>
     </>
   );
 };
 
-const styles = StyleSheet.create({
-  container: { flex: 1 },
-  view: {
-    ...StyleSheet.absoluteFillObject,
-    zIndex: Z_INDEX.IN_FRONT,
-  },
-});
+const useStyles = () => {
+  const { theme } = useTheme();
+  return useMemo(
+    () =>
+      StyleSheet.create({
+        mainContainer: { flex: 1 },
+        callContent: { flex: 1 },
+        safeArea: { flex: 1 },
+        unsafeArea: {
+          position: 'absolute',
+          left: 0,
+          right: 0,
+          bottom: 0,
+          // TODO: should be replaced with safeAreaInsets.bottom for all devices on ios and android
+          height: 34,
```

Comment:
@santhoshvai what do you think about using the `react-native-safe-area-context` in the SDK? We use it in the sample apps, but I think we need it in the sdk as well.

Reply:
strict NO here

the components can go inside bottom tabs, inside headers and what not

we should not freely assume that its always inside a whole screen

Reply:
ideally if wrapped under safe area the padding is applied somehow

for example, padding is passed through theme or props

Reply:
Great points! It makes complete sense, I will try to add it to theme or passable by props by integrator 👍

Reply:
I added the insets in the theme with this [commit](https://github.com/GetStream/stream-video-js/pull/1514/commits/2495166a054ae4f91aaf1a87eae792c911349e94). If you approve the solution I will create a cookbook for this.